src/App.tsx

```diff
@@ -0,0 +1,73 @@
+import React, { Component } from 'react';
+import Auth from './containers/AuthForm/Auth';
+import { Redirect, Route, Switch } from 'react-router';
+import { NavLink } from 'react-router-dom';
+import Button from './components/UI/Button/Button';
+import { Dispatch } from 'redux';
+import { logOut } from './store/auth/actions';
+import { connect } from 'react-redux';
+import { RootState } from './store';
+import { Fragment } from 'react';
+
+interface StateProps {
+  isAuthenticated: boolean
+}
+
+interface DispatchProps {
+  onLogOut: () => void
+}
+
+type Props = StateProps & DispatchProps
+
+class App extends Component<Props> {
+
+  logOutHandler = () => {
+    this.props.onLogOut();
+  };
+
+  render() {
+    // ToDo: Create a navigation component
+
+    const content = () => {
```

Comment:
Hint: this could be refactored into a private field, as it is already lazy and doesn't use any variables from the closure

Reply:
This component was rewritten in a FC style)